localization/ekf_localizer/src/ekf_localizer.cpp

```diff
@@ -428,13 +430,8 @@ void EKFLocalizer::measurementUpdatePose(const geometry_msgs::msg::PoseWithCovar
   delay_time = std::max(delay_time, 0.0);
 
   int delay_step = std::roundf(delay_time / ekf_dt_);
-  if (delay_step > params_.extend_state_step - 1) {
-    warning_.warnThrottle(
-      fmt::format(
-        "Pose delay exceeds the compensation limit, ignored. delay: %f[s], limit = "
-        "extend_state_step * ekf_dt : %f [s]",
-        delay_time, params_.extend_state_step * ekf_dt_),
-      1000);
+  if (!delayStepIsSufficientlySmall(delay_step, params_.extend_state_step)) {
```

Comment:
Do we still need this function? How about this?
```suggestion
  if (delay_step >= params_.extend_state_step) {
```

Reply:
That's true. So only the difference is whether we can test it or not.

Reply:
I see, thank you!
But then, I doubt the necessity of adding a new test script just for this simple inequality. Of course it will be nice in terms of coverage, but then we may going to have a number of test scripts.

Reply:
How about keeping it this time and modify when we need to change or when we can cover it in another test?

Reply:
> How about keeping it this time

Is you suggestion to
- keep this line as `if (delay_step >= params_.extend_state_step) {` in this PR
- isolate this line as another function to cover it in a test, if needed in future (e.g. the function gets more complicated)

? Then yes, I agree

Reply:
I mean I want coverage so keeping `delayStepIsSufficientlySmall` and its test, then in the future if we can isolate some parts in `measurementUpdatePose` including `delayStepIsSufficientlySmall`, we can safely remove `delayStepIsSufficientlySmall` and its test.

Reply:
Hmm, I believe we should revert the `delayStepIsSufficientlySmall` function, since 
- the advantage of testing the inequality operator of int seems small
- it decreases the readability IMHO (we should look into another file just to see the simple inequality operation)

I understand the importance of test coverage, but in this particular case I think we don't need an additional test.

Reply:
Resolved in [78875d0](https://github.com/autowarefoundation/autoware.universe/pull/2657/commits/78875d0008c9132af525cd87ac4bd1d7e95fc68f)

Reply:
Thank you!